packages/compass-connections/src/provider.ts

```diff
@@ -9,6 +9,7 @@ export type { DataService };
 export * from './connections-manager';
 export { useConnections } from './stores/connections-store';
 export { useActiveConnections } from './hooks/use-active-connections';
+export { isOIDCAuth } from './connections-manager';
```

Comment:
Let's not make this part of the public interface for this package if it's not used anywhere else? I'd almost be inclined to add a TODO comment to its definition to remove it, at least the usage in `connections-manager.ts` shouldn't really be necessary anymore

Reply:
I'm not sure I follow, it is required at least here:

```
const isOIDCConnectionAttempt = isOIDCAuth(
        connectionInfo.connectionOptions.connectionString
      );

      const adjustedConnectionInfoForConnection: ConnectionInfo = merge(
        cloneDeep(connectionInfo),
        {
          connectionOptions: adjustConnectionOptionsBeforeConnect({
            connectionOptions: connectionInfo.connectionOptions,
            defaultAppName: appName,
            preferences: { forceConnectionOptions, browserCommandForOIDCAuth },
            notifyDeviceFlow: !isOIDCConnectionAttempt
              ? undefined
              : onNotifyOIDCDeviceFlow ??
                (() => {
                  return;
                }),
          }),
        }
      );
```

That only enables the callback for `notifyDeviceFlow` for OIDC auth. Do you think this should be enabled always?

Reply:
Yeah, so, like, what is the bad thing that happens if this is just `notifyDeviceFlow: onNotifyOIDCDeviceFlow`? (Genuine question, I haven't checked, and this is not blocking for this PR anyway, just wanted to clarify whether this particular function really needs to be exported)

Reply:
I see the usage in the connection-store to modify the connection modal message, we probably can limit the usage to that and remove this check from here, nothing seems to change if it's always passed here when testing locally (and it's undefined or a noop in some cases here anyway so we already kinda do this)